client/src/client.rs

```diff
@@ -1017,6 +1017,21 @@ pub trait RpcApi: Sized {
         self.call("uptime", &[])
     }
 
+    /// Submit a raw block
+    fn submit_block(&self, block: &[u8]) -> Result<String> {
+        let block_hex: String = block.to_hex();
+        self.submit_block_hex(&block_hex)
+    }
+
+    /// Submit a block as a hex string
+    fn submit_block_hex(&self, block_hex: &String) -> Result<String> {
+        match self.call("submitblock", &[into_json(&block_hex)?]) {
+            Ok(serde_json::Value::Null) => Ok("Success".to_string()),
+            Ok(res) => Ok(res.to_string()),
+            Err(err) => Err(err),
```

Comment:
You could use `?` on `call()` to simplify handling the `Err` case, something like `match self.call(...)? { serde_json::Value::Null => Ok(..), res => Err(...) }`

Reply:
Using `match self.call(...)?` for me leads to a type inference error ('cannot infer type for type parameter `E` declared on the enum `Result`'). Even if there are ways around that by manually specifying the type argument, is `Err(err) => Err(err),` not much simpler to read and understand?

Reply:
Sounds good to me :+1: